app/questionnaire/placeholder_parser.py

```diff
@@ -178,6 +180,23 @@ def _parse_transforms(
 
                 transform_args[arg_key] = resolved_value
 
+                if (
+                    transform["transform"] == "format_currency"
+                    and arg_value.get("source") == "answers"
+                ):
+                    transform_args["decimal_limit"] = self._schema.get_decimal_limit(
+                        [arg_value["identifier"]]
+                    )
+
+                if (
+                    transform["transform"] == "format_currency"
+                    and arg_value.get("source") == "calculated_summary"
+                ):
+                    answer_ids = get_calculated_summary_answer_ids(self._schema.get_block(arg_value["identifier"]))  # type: ignore
+                    transform_args["decimal_limit"] = self._schema.get_decimal_limit(
+                        answer_ids
+                    )
```

Comment:
We should review this implementation as this isn't pretty fragile. We should avoid hard coding logic like this in general. We might have other cases in future in which based on the argument, we need to do custom logic, We should leave it up to the transform to handle any logic special to them. Placeholder parser should be generic. It should be done similar to `TRANSFORMS_REQUIRING_ROUTING_PATH`. 

What is essentially happening is, some transforms might need to use the raw arguments to work out some info needed for the final transform. We could pass in these unresolved argument to the transform then the transform can do whatever special logic it wants. It means each transform handles any special logic to them and not placeholder parser. I suggest introducing a constant like `TRANSFORMS_REQUIRING_UNRESOLVED_ARGUMENTS` then if transform name is in them, it should also pass along the raw arguments which the transform can use to handle the special cases.

If anything is not clear let me know.

Reply:
Updated this 👍